api/tests/core/fraud/test_api.py

```diff
@@ -130,19 +131,16 @@ def test_create_profile_completion_fraud_check(self, caplog):
         }
 
 
-def filter_invalid_fraud_items_to_reason_code(
+def filter_invalid_fraud_items_to_reason_codes(
     fraud_items: list[fraud_models.FraudItem],
-) -> list[fraud_models.FraudItem]:
-    return [item.reason_code for item in fraud_items if item.status != fraud_models.FraudStatus.OK]
+) -> set[fraud_models.FraudReasonCode]:
+    return set(
+        itertools.chain(*[item.reason_codes for item in fraud_items if item.status != fraud_models.FraudStatus.OK])
```

Comment:
(détail / style) Tu peux utiliser [chain.from_iterable()](https://docs.python.org/3/library/itertools.html#itertools.chain.from_iterable) : 

```suggestion
        itertools.chain.from_iterable(item.reason_codes for item in fraud_items if item.status != fraud_models.FraudStatus.OK)
```